spring-boot-project/spring-boot/src/main/java/org/springframework/boot/webservices/client/WebServiceTemplateBuilder.java

```diff
@@ -0,0 +1,1106 @@
+	 * @param checkConnectionForError Specify whether checkConnectionForError should be
+	 * enabled or not.
+	 * @return a new builder instance.
+	 **/
+
+	public WebServiceTemplateBuilder setCheckConnectionForError(
+			boolean checkConnectionForError) {
+		return new WebServiceTemplateBuilder(this.interceptors,
+				append(this.internalCustomizers,
+						new CheckConnectionForErrorCustomizer(checkConnectionForError)),
+				this.customizers, this.webServiceMessageSenderSuppliers,
+				this.webServiceMessageSenderCustomizers, this.marshaller,
+				this.unmarshaller, this.destinationProvider, this.transformerFactoryClass,
+				this.messageFactory, this.detectWebServiceMessageSender);
+	}
+
+	/**
+	 * Sets the {@code Supplier} of {@link WebServiceMessageSender} that should be called
+	 * each time when {@link #configure(WebServiceTemplate)} method is called.
+	 *
+	 * <b>Note!</b> Override existing WebServiceMessageSender {@code suppliers}
+	 * @param webServiceMessageSenderSupplier Supplier for the messageSender
+	 * @return a new builder instance.
+	 * @see WebServiceTemplate#setMessageSenders(WebServiceMessageSender[])
+	 * @see #setWebServiceMessageSenders(Collection)
+	 */
+
+	public WebServiceTemplateBuilder setWebServiceMessageSender(
+			Supplier<? extends WebServiceMessageSender> webServiceMessageSenderSupplier) {
+		Assert.notNull(webServiceMessageSenderSupplier,
+				"webServiceMessageSenderSupplier must not be null");
+		return setWebServiceMessageSenders(
+				Collections.singleton(webServiceMessageSenderSupplier));
+	}
+
+	/**
+	 * Add additional {@code Supplier} of {@link WebServiceMessageSender} that should be
+	 * called each time when {@link #configure(WebServiceTemplate)} method is called.
+	 * @param webServiceMessageSenderSupplier Supplier for the messageSender
+	 * @return a new builder instance.
+	 * @see WebServiceTemplate#setMessageSenders(WebServiceMessageSender[])
+	 * @see #addWebServiceMessageSenders(Collection)
+	 */
+	public WebServiceTemplateBuilder addWebServiceMessageSender(
+			Supplier<? extends WebServiceMessageSender> webServiceMessageSenderSupplier) {
+		Assert.notNull(webServiceMessageSenderSupplier,
+				"webServiceMessageSenderSupplier must not be null");
+		return addWebServiceMessageSenders(
+				Collections.singleton(webServiceMessageSenderSupplier));
+	}
+
+	/**
+	 * Sets the {@code Class} of {@link WebServiceMessageSender} that should be created
+	 * each time when {@link #configure(WebServiceTemplate)} method is called.
+	 *
+	 * <b>Note!</b> Override existing WebServiceMessageSender {@code suppliers}
+	 * @param webServiceMessageSenderClass {@code Class} of
+	 * {@link WebServiceMessageSender}
+	 * @return a new builder instance.
+	 * @see WebServiceTemplate#setMessageSenders(WebServiceMessageSender[])
+	 * @see #setWebServiceMessageSender(Supplier)
+	 * @see BeanUtils#instantiateClass(Class)
+	 */
+
+	public WebServiceTemplateBuilder setWebServiceMessageSender(
```

Comment:
there are too many flavours to set a `WebServiceMessageSender`. Specifying a vararg of instance and a `Class` looks wrong to me. I've removed that in my fork.

Reply:
the was an idea to have a class instead of a supplier.

Reply:
Got it but I think the reason why a `Supplier` was added is wrong.